.changeset/clean-kiwis-boil.md

```diff
@@ -0,0 +1,5 @@
+---
+'react-textarea-autosize': patch
+---
+
+replace useEffect with useLayoutEffect for React 17
```

Comment:
Moved internal `'resize'` listener to the layout effect since React 17 calls cleanups of regular effects asynchronously. This ensures that we don't ever try to access the already unmounted ref in our listener.